src/battle-scene-interface.ts

```diff
@@ -0,0 +1,176 @@
+interface IBattleScene {
```

Comment:
Explicitly defining an interface at all seems unnecessary. TypeScript supports duck typing, after all. You can just do `scene: BattleScene | BattleSceneStub` instead of `scene: IBattleScene`